python/python/bystro/ancestry/inference.py

```diff
@@ -266,16 +309,15 @@ def infer_ancestry(ancestry_models: AncestryModels, genotypes: Dataset) -> Ances
                 psutil.Process(os.getpid()).memory_info().rss / 1024**2,
             )
 
-            missing_rows = list(set(ancestry_model.pca_loadings_df.index) - set(genotypes_df.index))
-            if missing_rows:
-                missing_rows_df = pd.DataFrame(
-                    np.ones((len(missing_rows), len(genotypes_df.columns))),
-                    index=missing_rows,
-                    columns=genotypes_df.columns,
+            if DEBUG and DEBUG_MISSINGNESS_IMPOSED > 0:
+                # drop a random DEBUG_MISSINGNESS_IMPOSED number of rows
+                genotypes_df = genotypes_df.sample(
+                    frac=1 - DEBUG_MISSINGNESS_IMPOSED, random_state=42, axis=0
                 )
-                genotypes_df = pd.concat([genotypes_df, missing_rows_df])
 
-            genotypes_df[genotypes_df.isna() | (genotypes_df < 0)] = 1
+                logger.info(
+                    "DEBUG: imposed missingness, genotypes_df shape after: %s", genotypes_df.shape
+                )
```

Comment:
Recommend to add update to _make_trivial_probability_interval to account for small rounding errors